tests/TestCIT_KCI.py

```diff
@@ -0,0 +1,296 @@
+                          0.588, 0.753, 0.567, 0.754, 0.588, 0.736, 0.608, 0.754, 0.588, 0.748, 0.565, 0.754, 0.588,
+                          0.77, 0.594, 0.754, 0.588, 0.767, 0.616, 0.754, 0.588, 0.747, 0.584, 0.754, 0.588, 0.741,
+                          0.58, 0.754, 0.754, 0.755, 0.752, 0.754, 0.754, 0.737, 0.757, 0.754, 0.754, 0.762, 0.745,
+                          0.754, 0.754, 0.74, 0.749, 0.754, 0.754, 0.757, 0.751, 0.754, 0.754, 0.742, 0.752, 0.754,
+                          0.754, 0.753, 0.766, 0.754, 0.754, 0.753, 0.746, 0.754, 0.754, 0.75, 0.77, 0.754, 0.754,
+                          0.756, 0.746, 0.754, 0.754, 0.738, 0.753, 0.754, 0.754, 0.766, 0.749, 0.754, 0.754, 0.744,
+                          0.74, 0.754, 0.754, 0.765, 0.725, 0.754, 0.754, 0.753, 0.766, 0.754, 0.754, 0.753, 0.749,
+                          0.754, 0.754, 0.75, 0.759, 0.754, 0.754, 0.744, 0.771]
+        pvalue03_truth = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.138, 0.138, 0.143, 0.113,
+                          0.138, 0.138, 0.141, 0.15, 0.051, 0.051, 0.066, 0.057, 0.051, 0.051, 0.056, 0.048, 0.004,
+                          0.004, 0.006, 0.002, 0.004, 0.004, 0.004, 0.015, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+                          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+        pvalue032_truth = [0.133, 0.133, 0.119, 0.126, 0.142, 0.142, 0.142, 0.134, 0.133, 0.133, 0.121, 0.128, 0.142,
+                           0.142, 0.132, 0.13, 0.133, 0.133, 0.131, 0.12, 0.142, 0.142, 0.141, 0.135, 0.12, 0.12,
+                           0.114, 0.116, 0.116, 0.116, 0.109, 0.108, 0.12, 0.12, 0.105, 0.105, 0.116, 0.116, 0.105,
+                           0.101, 0.12, 0.12, 0.108, 0.108, 0.116, 0.116, 0.108, 0.105, 0.504, 0.504, 0.494, 0.518,
+                           0.005, 0.005, 0.005, 0.005, 0.477, 0.477, 0.473, 0.479, 0.076, 0.076, 0.076, 0.077, 0.549,
```

Comment:
hmmm, 0.005 seems weird. so small

Reply:
This and the above case are two cases that the Matlab itself will give weird results. The details are listed in [Weird_test_cases.pdf](https://github.com/cmu-phil/causal-learn/files/9157156/test_cases.pdf). 
I‘ve turned to Mingming for help and waiting for the answer. Hope I can give you the answer before this weekend.

Reply:
Thanks, that's super clear. I think we can leave it as it is for now, and after Kun and Mingming respond, we can change it if necessary.

Reply:
Super nice!! Thanks!!